pkgs/development/libraries/tomlplusplus/default.nix

```
@@ -0,0 +1,28 @@
+{ lib
+, stdenv
+, meson
+, ninja
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tomlplusplus";
+  version = "3.2.0";
+
+  src = fetchFromGitHub {
+    owner = "marzer";
+    repo = pname;
+    rev = "4b166b69f28e70a416a1a04a98f365d2aeb90de8";
```

Since there's a tag and it matches this could be

```suggestion
    rev = "v${version}";
```

---

But tags could be replaced after the fact. Wouldn't that break reproducability?

---

If the tag gets replaced the build will fail, there's a sha hash below.

By convention tags are usually used in nixpkgs, makes it easier to verify the commit's the right one. You can set a commit that's in a fork or a PR for github repos and it'll still fetch, so it's a potential vector for someone to sneak something in. I checked the tag and commit match here though.

---

Oh I should have looked at the code, not just the suggestion :sweat_smile: